src/disambiguate-series-statements.js

```diff
@@ -0,0 +1,218 @@
+import createDebugLogger from 'debug';
+import {fieldHasSubfield, fieldToString, nvdebug} from './utils';
+import {MARCXML} from '@natlibfi/marc-record-serializers';
+import {Error} from '@natlibfi/melinda-commons';
+import {default as createNatlibfiSruClient} from '@natlibfi/sru-client';
+
+//const {default: createNatlibfiSruClient} = natlibfiSruClient;
+
+const debug = createDebugLogger('@natlibfi/marc-record-validators-melinda:disambiguate-series-statements');
+
+const ELECTRONIC = 1;
+const PRINTED = 2;
+const UNDEFINED = 0;
+const SRU_API_URL = 'https://sru.api.melinda.kansalliskirjasto.fi/bib';
+
+// Author(s): Nicholas Volk
+export default function () {
+  const sruClient = createSruClient(SRU_API_URL);
+
+  return {
+    description: 'Disambiguate between printed and electonic series statements (field 490)',
+    validate, fix
+  };
+
+  async function fix(record) {
+    //console.log(`fix(record)`); // eslint-disable-line no-console
+    const deletableFields = await getDeletableFields(record);
+
+    const deletableFieldsAsStrings = deletableFields.map(f => fieldToString(f));
+
+    deletableFields.forEach(nf => nvdebug(`TODO: remove field '${fieldToString(nf)}'`, debug));
+
+    deletableFields.forEach(f => record.removeField(f));
+
+    return {message: [], fix: deletableFieldsAsStrings, valid: true};
+  }
+
+  async function validate(record) {
+    const deletableFields = await getDeletableFields(record);
+
+    if (deletableFields.length === 0) {
+      return {'message': [], 'valid': true};
+    }
+
+    const deletableFieldsAsStrings = deletableFields.map(f => `REMOVE '${fieldToString(f)}'`);
+
+    return {'message': deletableFieldsAsStrings, 'valid': false};
+  }
+
+  async function getDeletableFields(record) {
+    const recordType = getRecordType(record);
+    //console.info(`getDeletableFields(record), type: ${recordType}`); // eslint-disable-line no-console
+    if (recordType === UNDEFINED) {
+      return [];
+    }
+
+    const seriesStatements = record.get('490').filter(f => fieldHasSubfield(f, 'x'));
+
+    //console.info(`N STATEMENTS=${seriesStatements.length}`); // eslint-disable-line no-console
+    if (seriesStatements.length < 2) {
```

Comment:
This blocks handling of case where there are two $x within one f490

Reply:
Outdated. My bad.